File: LegacySource/settlement/room/tests/production/Input.cs
using init.resources;

namespace settlement.room.tests.production
{
    public class Input
    {
        public readonly RESOURCE res;
        public readonly double amount;
        public readonly TestRecipe producer;

        public Input(double amount, TestRecipe producer)
        {
            this.res = producer.res;
            this.amount = amount;
            this.producer = producer;
        }

        public double wTot(ProductionSpec ibonuses)
        {
            return amount * producer.wTotPerItem(ibonuses);
        }
    }
}

File: settlement/room/tests/production/Input.java
package settlement.room.tests.production;

import init.resources.RESOURCE;

public class Input {

	public final RESOURCE res;
	public final double amount;
	public final TestRecipe producer;

	Input(double amount, TestRecipe producer) {
		this.res = producer.res;
		this.amount = amount;
		this.producer = producer;
	}

	double wTot(ProductionSpec ibonuses) {
		return amount * producer.wTotPerItem(ibonuses);
	}

}